### PR TITLE
fix multiple aliases for same field results in postgres error (fix #278)

### LINKED ISSUE #278: aliasing same relationship multiple times in a query results in postgres error
<details>
<summary>Query</summary>

```graphql
query {
  items {
    id
    value
    related_data: related_table_values {
      data
    }
    related_id: related_table_values {
      id
    }
  }
}
```

</details>

<details>
<summary>Response</summary>

```json
{
  "errors": [
    {
      "internal": {
        "statement": "SELECT  coalesce(json_agg((SELECT  \"e\"  FROM  (SELECT  \"r\".\"related_table_values\" AS \"related_data\", \"r\".\"value\" AS \"value\", \"r\".\"id\" AS \"id\", \"r\".\"related_table_values\" AS \"related_id\"       ) AS \"e\"      ) ), '[]' )  FROM  (SELECT  CASE WHEN (hdb_catalog.first(\"r\".\"__r_related_table_values_item_id\" )) IS NULL THEN '[]' ELSE json_agg((SELECT  \"e\"  FROM  (SELECT  \"r\".\"data\" AS \"data\"       ) AS \"e\"      ) ) END AS \"related_table_values\", hdb_catalog.first(\"l\".\"related_table_values\" ) AS \"related_table_values\", hdb_catalog.first(\"l\".\"value\" ) AS \"value\", hdb_catalog.first(\"l\".\"id\" ) AS \"id\", hdb_catalog.first(\"l\".\"__l_related_table_values_id\" ) AS \"__l_related_table_values_id\", hdb_catalog.first(\"l\".\"__l_related_table_values_id\" ) AS \"__l_related_table_values_id\" FROM  (SELECT  CASE WHEN (hdb_catalog.first(\"r\".\"__r_related_table_values_item_id\" )) IS NULL THEN '[]' ELSE json_agg((SELECT  \"e\"  FROM  (SELECT  \"r\".\"id\" AS \"id\"       ) AS \"e\"      ) ) END AS \"related_table_values\", hdb_catalog.first(\"l\".\"value\" ) AS \"value\", hdb_catalog.first(\"l\".\"id\" ) AS \"id\", hdb_catalog.first(\"l\".\"__l_related_table_values_id\" ) AS \"__l_related_table_values_id\", hdb_catalog.first(\"l\".\"__l_related_table_values_id\" ) AS \"__l_related_table_values_id\" FROM  (SELECT  \"value\" AS \"value\", \"id\" AS \"id\", \"id\" AS \"__l_related_table_values_id\", \"id\" AS \"__l_related_table_values_id\" FROM \"public\".\"items\"  WHERE (('true') AND (('true') AND ('true')))     ) AS \"l\" LEFT OUTER JOIN LATERAL (SELECT  \"item_id\" AS \"__r_related_table_values_item_id\", \"id\" AS \"id\" FROM \"public\".\"item_related_table\"  WHERE ((((\"l\".\"__l_related_table_values_id\") = (\"item_id\")) AND ('true')) AND (('true') AND ('true')))     ) AS \"r\" ON ('true')  GROUP BY \"l\".\"__l_related_table_values_id\"    ) AS \"l\" LEFT OUTER JOIN LATERAL (SELECT  \"item_id\" AS \"__r_related_table_values_item_id\", \"data\" AS \"data\" FROM \"public\".\"item_related_table\"  WHERE ((((\"l\".\"__l_related_table_values_id\") = (\"item_id\")) AND ('true')) AND (('true') AND ('true')))     ) AS \"r\" ON ('true')  GROUP BY \"l\".\"__l_related_table_values_id\"    ) AS \"r\"      ",
        "prepared": true,
        "error": {
          "exec_status": "FatalError",
          "hint": null,
          "message": "column reference \"__l_related_table_values_id\" is ambiguous",
          "status_code": "42702",
          "description": null
        },
        "arguments": []
      },
      "path": "$",
      "error": "postgres query error",
      "code": "postgres-error"
    }
  ]
}
```

</details>

[via arpitjacob#4840 on discord]

### PR DESCRIPTION
**Pull request by rakeshkky**

### CODE REVIEW

**Comment by revskill10** [comment]
@rakeshkky 
This query works:
```
query{
  article1:article(limit:1){
    author_id
    content
  }
  article2:article(limit:2){
    author_id
    content
  }
}
```
But if you replace query with subscription, it failed.
The error is:
```
anonymous subscription must select only one top level field
```

**Comment by 0x777** [comment]
This is blocked on https://github.com/hasura/graphql-engine/issues/6

File: server/src-lib/Hasura/GraphQL/Resolve/Select.hs [hunk 1]
```diff
@@ -42,11 +42,13 @@ fromSelSet fldTy flds =
...
         case fldInfo of
-          Left (PGColInfo pgCol colTy _) -> return (rqlFldName, RS.FCol (pgCol, colTy))
+          Left (PGColInfo pgCol colTy _) -> do
+            let colAlias = S.Alias $ Iden $ getFieldNameTxt rqlFldName
+            return (rqlFldName, RS.FCol (pgCol, colTy, Just colAlias))
           Right (relInfo, tableFilter, tableLimit, _) -> do
             let relTN = riRTable relInfo
             relSelData <- fromField relTN tableFilter tableLimit fld
-            let annRel = RS.AnnRel (riName relInfo) (riType relInfo)
+            let annRel = RS.AnnRel (RelName $ getFieldNameTxt rqlFldName) (riType relInfo)
```
Comment (nizar-m) [inline server/src-lib/Hasura/GraphQL/Resolve/Select.hs]: Use data type `Alias` instead of `RelName`

**Comment by 0x777** [comment]
closing in favor of #643

**Pull request closed without merging**